### PR TITLE
Add ToolStripItemTarget.

### PR DESCRIPTION
**Pull request by Fr33dan**
After encountering NLog main project "bug" [#147](https://github.com/NLog/NLog/issues/147). I implemented a target capable of logging to a ToolStripItem. The documentation is in the main project under pull request [#4566](https://github.com/NLog/NLog/pull/4566)

### CODE REVIEW

**Comment by 304NotModified** [comment]
Thanks! Will review soon!

File: NLog.Windows.Forms/ToolStripItemTarget.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        }
+
+        private delegate void DelSendTheMessageToFormControl(ToolStripItem control, string logMessage);
+
+        /// <summary>
+        /// Gets or sets the name of the ToolStripItem to which NLog will log write log text.
+        /// </summary>
+        /// <docgen category='Form Options' order='10' />
+        [RequiredParameter]
+        public string ItemName { get; set; }
```
Comment (304NotModified) [inline NLog.Windows.Forms/ToolStripItemTarget.cs:48]: maybe nice to make all the properties layoutable?

**Review by 304NotModified — CHANGES_REQUESTED** [review]
looks good, but I need some unit tests so we could keep it working

File: NLog.Windows.Forms/ToolStripItemTarget.cs (hunk 1 above)
Comment (Fr33dan) [reply NLog.Windows.Forms/ToolStripItemTarget.cs:48]: I'm not exactly sure how you mean layoutable. Is there an example of a layoutable property in the project I can use an an example to understand what you mean?

Comment (304NotModified) [reply NLog.Windows.Forms/ToolStripItemTarget.cs:48]: Use the type `Layout` instead of `string`, and use .Render(logEvent).

**Comment by Fr33dan** [comment]
I also did as suggested and made the values Layouts instead of strings so you can use Layout to create a more advanced configuration.

I used FormControlTarget as a template, but when making the ToolStripItemTarget itself. Unfortunately it doesn't look like there are unit tests for that for me to use as a template so I made up what I thought was best.At the moment it's not working however and I'm not sure entirely sure why. If I call the test method from my desktop it works, but fails in the appveyor testing.

I think it's some kind of threading issue but I've approached every angle I can think of on that front at the moment and I've had enough fighting with it for today. If anyone has any suggestions for when I get back at that would be appreciated.

**Comment by Fr33dan** [comment]
Figured it out! Because there was only one class of tests originally XUnit was putting them all in the same collection and running them in sequence. When I added a new class that was a different collection by default causing them to run in parallel. Two threads trying to run UI threads simultaneously causes problems.

This explains why one of the existing RichTextBoxTests was failing despite me not changing that code at all. I added a collections attribute to both classes to make them all part a collection named "NLog.Windows.Forms Tests". Now all tests run in sequence and it passes. 

If in the future unit tests are added for the FormControlTarget they will need to also be part of this Collection.

**Comment by 304NotModified** [comment]
Thanks!

**Comment by snakefoot** [comment]
Thank you for the wiki-page: https://github.com/NLog/NLog/wiki/ToolStripItem-Target

**Pull request merged**